Dockerfile

```diff
@@ -1,10 +1,20 @@
+# Configurable arguments
 ARG BUILD_ENV=default
-FROM golang:1.16-buster AS base
+ARG GOLANG_VERSION=1.16-buster
+
+# Install build time dependencies
+FROM golang:${GOLANG_VERSION} AS base
+
+ARG APT_INSTALL_EXTRA_DEPS
+ENV APT_INSTALL_EXTRA_DEPS=$APT_INSTALL_EXTRA_DEPS
 
 RUN apt-get update && \
   apt-get install --no-install-recommends --assume-yes curl unzip && \
   apt-get clean
 
+RUN echo ${APT_INSTALL_EXTRA_DEPS}
+RUN apt-get install --no-install-recommends --assume-yes $APT_INSTALL_EXTRA_DEPS
```

Comment:
Just checking this doesn't error if APT_INSTALL_EXTRA_DEPS is empty?

Reply:
running just ```apt-get install``` with no other argument does just a state check of the existing apt packages